eng/cake/dotnet.cake

```diff
@@ -97,12 +97,25 @@ Task("VS-NET6")
 Task("VS-WINUI")
     .Description("Provisions .NET 6 and launches an instance of Visual Studio with WinUI projects.")
     .IsDependentOn("Clean")
-    .IsDependentOn("dotnet")
-    .IsDependentOn("dotnet-buildtasks")
+    // .IsDependentOn("dotnet") WINUI currently can't launch application with local dotnet 
+    // .IsDependentOn("dotnet-buildtasks")
     .Does(() =>
     {
-        RunMSBuildWithLocalDotNet("./Microsoft.Maui.WinUI.sln");
-        StartVisualStudioForDotNet6("./Microsoft.Maui.WinUI.sln");
+        string sln = "./Microsoft.Maui.WinUI.sln";
+        var msbuildSettings = new MSBuildSettings
+        {
+            Configuration = configuration,
+            ToolPath = FindMSBuild(),
+        };
+
+        MSBuild("./Microsoft.Maui.BuildTasks-net6.sln", msbuildSettings.WithRestore());
+        MSBuild(sln, msbuildSettings.WithRestore());
+
+        var vsLatest = VSWhereLatest(new VSWhereLatestSettings { IncludePrerelease = true, });
+        if (vsLatest == null)
+            throw new Exception("Unable to find Visual Studio!");
+        Environment.SetEnvironmentVariable("_ExcludeMauiProjectCapability", "true", EnvironmentVariableTarget.Process);
```

Comment:
Is there a reason opening VS in the regular way doesn't work?

Do you need `$(_ExcludeMauiProjectCapability)` for WinUI projects?

Reply:
> Is there a reason opening VS in the regular way doesn't work?

Regular way? I just have it open from cake with the correct SLN so once it's done you're ready to go.

> Do you need $(_ExcludeMauiProjectCapability) for WinUI projects?

I wasn't sure what all this did so I just left it because I figured it wasn't harming anything but I can remove it

Reply:
"The regular way" would just be opening VS yourself and not running cake. Mainly asking if you need this target at all.

Reply:
Yea, this target ensures that the Build Tasks solution gets built which is needed for a lot of different things (Xaml/Resizetizer) 

If you just open the WinUI sln by itself then you have to go through and build all the tasks projects and a few other things before you get to a runnable state